specs/latest/1.0/index.html

```diff
@@ -2653,10 +2671,23 @@ <h4>Texture objects</h4>
             The width and height of the texture are set as specified in section
             <a href="#TEXTURE_UPLOAD_SIZE">Texture Upload Width and Height</a>.<br><br>
 
-            The source image data is conceptually first converted to the data type and format
-            specified by the <em>format</em> and <em>type</em> arguments, and then transferred to
-            the WebGL implementation. Format conversion is performed according to the following table.
-            If a packed pixel format is specified which would imply loss of bits of precision from the
+            First, the source image data is conceptually converted to the color space
+            specified by the <a href="#DOM-WebGLRenderingContext-colorSpace">colorSpace</a>
+            attribute, except in the following situations:
+
+            <ul>
+            <li> the source image data is an <code>HTMLImageElement</code>, and
+                 the <code>UNPACK_COLORSPACE_CONVERSION_WEBGL</code> pixel storage
+                 parameter is set to <code>NONE</code>. </li>
+            <li> the source image data is an <code>ImageBitmap</code>, in which case
```

Comment:
I think the scheme will be to have ImageBitmapOptions have an optional colorSpace parameter. If the parameter was specified, and matches the rendering context's color space, then the conversion is a no-op by definition.

Reply:
(so I'd remove this second exception, or make it non-normative)

Reply:
Just to check, there wouldn't be any special cases for ImageBitmaps created _without_ the optional colorSpace parameter set, right? I.e. they'll always be tagged with a particular colorspace (srgb by default I think)

> If the parameter was specified, and matches the rendering context's color space, then the conversion is a no-op by definition.

This is also what I've done in WebGPU, including the non-normative notes. https://gpuweb.github.io/gpuweb/#gpu-image-copy-texture-tagged

Reply:
ImageBitmaps created without a colorSpace won't be special (they'll be just like, say, an Image).

If an ImageBitmap is created without the colorSpace parameter specified, it will have its own internal color space (e.g, specified by an ICC profile, or video properties, etc). 
* There is no interface through which the actual pixel values (and their color space) can be queried.
* If that ImageBitmap is put into an ImageBitmapRenderingContext, it will appear in full fidelity exactly as it would if its source were put into an element (e.g, a rec2020, HLG-HDR video's frame will appear as rec2020 HLG-HDR, etc).
* If that ImageBitmap is imported into WebGL/GPU, it will be converted to the target color space at the moment of that import.

Let me know if this didn't hit the mark.

Reply:
All sounds right to me, I just wasn't sure if it would be in the source space or in srgb if no conversion options were set. Source space does seem right, and I think there are still reasonable knobs apps can set to avoid unnecessary conversion steps.

Reply:
Per initial prototyping in Chromium in https://chromium-review.googlesource.com/c/chromium/src/+/3560989 - this section needs to be updated. It turns out it is beneficial to perform the colorspace conversion for ImageBitmaps as late as possible, during uploading to the WebGL context, and taking into account `unpackColorSpace`.

Reply:
Have updated this PR to specify that ImageBitmaps are affected by unpackColorSpace, and provided rationale.